File: Front/src/models/produtos-do-carrinho.ts
import { Produto } from "./produto";

export class ProdutosDoCarrinho {
  produto: Produto = {};
  quantidade?: number = 1;
  produtoId?: number = 0;

  constructor(produto: Produto, quantidade: number) {
    this.produto = produto;
    this.quantidade = quantidade;
    this.produtoId = produto.id;
  }
}

File: Back/Back/Models/ProdutosDoCarrinho.cs
namespace Back.Models;

public class ProdutosDoCarrinho : Entity
{
    public Produto Produto { get; private set; }
    public int ProdutoId { get; private set; }
    public int Quantidade { get; private set; }

    //EF Constructor
    protected ProdutosDoCarrinho()
    {
    }

    public ProdutosDoCarrinho(Produto produto, int produtoId, int quantidade)
    {
        Produto = produto;
        ProdutoId = produtoId;
        Quantidade = quantidade;
    }
}